pkg/auth/providers/saml/saml_provider.go

```
@@ -0,0 +1,237 @@
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/auth/providers/common"
+	"github.com/rancher/rancher/pkg/auth/tokens"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/client/management/v3"
+	publicclient "github.com/rancher/types/client/management/v3public"
+	"github.com/rancher/types/config"
+	"github.com/rancher/types/user"
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const PingName = "ping"
+
+type Provider struct {
+	Ctx             context.Context
+	AuthConfigs     v3.AuthConfigInterface
+	UserMGR         user.Manager
+	TokenMGR        *tokens.Manager
+	ServiceProvider *saml.ServiceProvider
+	Name            string
+	UserType        string
+	GroupType       string
+	ClientState     samlsp.ClientState
+}
+
+var SamlProviders = make(map[string]*Provider)
+
+func Configure(ctx context.Context, mgmtCtx *config.ScaledContext, userMGR user.Manager, tokenMGR *tokens.Manager, name string) common.AuthProvider {
+	samlp := &Provider{
+		Ctx:         ctx,
+		AuthConfigs: mgmtCtx.Management.AuthConfigs(""),
+		UserMGR:     userMGR,
+		TokenMGR:    tokenMGR,
+		Name:        name,
+		UserType:    name + "_user",
+		GroupType:   name + "_group",
+	}
+	SamlProviders[PingName] = samlp
+	return samlp
+}
+
+func (s *Provider) GetName() string {
+	return s.Name
+}
+
+func (s *Provider) CustomizeSchema(schema *types.Schema) {
+	schema.ActionHandler = s.actionHandler
+	schema.Formatter = s.formatter
+}
+
+func (s *Provider) TransformToAuthProvider(authConfig map[string]interface{}) map[string]interface{} {
+	p := common.TransformToAuthProvider(authConfig)
+	if s.Name == PingName {
+		p[publicclient.PingProviderFieldRedirectURL] = formSamlRedirectURLFromMap(authConfig, s.Name)
+	}
+	return p
+}
+
+func (s *Provider) AuthenticateUser(input interface{}) (v3.Principal, []v3.Principal, string, error) {
+	return v3.Principal{}, nil, "", nil
```

add a comment explaining why this is not implemented

---

in fact, lets return an error saying that this method is not supported